### PR TITLE
MPN-1002: complete exercises for Functional Programming in Scala, Chapter 6

### PR DESCRIPTION
**Pull request by yawaramin**

### CODE REVIEW

File: exercises/src/main/scala/fpinscala/state/State.scala [hunk 1]
```diff
@@ -30,32 +29,106 @@ object RNG {
...
+
+  def double3(rng: RNG): ((Double,Double,Double), RNG) = {
+    val (d1, rng2) = double(rng)
+    val (d2, rng3) = double(rng2)
+    val (d3, rng4) = double(rng3)
+
+    (d1, d2, d3) -> rng4
+  }
+
+  def ints(count: Int)(rng: RNG): (List[Int], RNG) = {
```
Comment (gpoirier) [inline exercises/src/main/scala/fpinscala/state/State.scala:75]: Did you consider using fold for this? Also, I didn't find 6.7 with the rewrite of this with `sequence`.

File: exercises/src/test/scala/state/StateSpec.scala [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+      prop { (xs: List[Int], s: Int) =>
+        State.sequence(xs.map(State.unit[Int, Int])).run(s)._1 mustEqual {
+          xs
+        }
+      }
+    }
+  }
+
+  "State.simulateMachine" should {
+    "succeed at running simulation in book" in {
```
Comment (gpoirier) [inline exercises/src/test/scala/state/StateSpec.scala:242]: What about edge cases? Out of order operations, empty operations, etc.

File: exercises/src/main/scala/fpinscala/state/State.scala (hunk 1 above)
Comment (yawaramin) [inline exercises/src/main/scala/fpinscala/state/State.scala:75]: @gpoirier I didn't consider using a fold; I felt it would be more natural to use an unfold or an iterate function to express this.

About the re-implementation, you're absolutely right--I'll do that now.

File: exercises/src/test/scala/state/StateSpec.scala (hunk 2 above)
Comment (yawaramin) [inline exercises/src/test/scala/state/StateSpec.scala:242]: @gpoirier Oops, you're right--I'll add tests for edge cases.

**Comment by gpoirier** [comment]
+1
